Data/HashMap/Base.hs

```diff
@@ -1822,6 +1799,56 @@ updateOrConcatWithKey f ary1 ary2 = A.run $ do
     return mary
 {-# INLINABLE updateOrConcatWithKey #-}
 
+-- Helper functions for very rare cases.
+
+-- Remove (any) one element from an array of hashmaps
+unConsA :: A.Array (HashMap k v) -> UnCons k v
+unConsA ary = case A.length ary of
+  0 -> WasEmpty
+  1 -> case A.index ary 0 of
+    Empty -> WasEmpty -- That would be weird, but fine.
```

Comment:
This all looks much too complicated. We maintain the invariant that an array *never* contains an empty map. In addition, this function as currently conceived doesn't seem likely to play well with single-element-array collapsing that I mention below.